lib/utils/github.ts

```diff
@@ -3,21 +3,31 @@
  * @todo Use `getAvatarById` instead of `getAvatarByUsername` whenever possible
  * @see {@link https://github.com/open-sauced/insights/issues/746}
  */
-const getAvatarByUsername = (username: string | null, size=460) =>
+const getAvatarByUsername = (username: string | null, size = 460) =>
   `https://www.github.com/${username ?? "github"}.png?size=${size}`;
 
-const getAvatarById = (id: string | null, size=460) =>
+const getAvatarById = (id: string | null, size = 460) =>
   `https://avatars.githubusercontent.com/u/${id ?? "9919"}?size=${size}&v=4`;
 
-const getProfileLink = (username: string | null) =>
-  `https://github.com/${username ?? ""}`;
+const getProfileLink = (username: string | null) => `https://github.com/${username ?? ""}`;
 
-const getRepoIssuesLink = (repoName: string | null) =>
-  `https://github.com/${repoName && `${repoName}/issues` || ""}`;
+const getRepoIssuesLink = (repoName: string | null) => `https://github.com/${(repoName && `${repoName}/issues`) || ""}`;
 
-export {
-  getAvatarById,
-  getAvatarByUsername,
-  getProfileLink,
-  getRepoIssuesLink
+const generateApiPrUrl = (url: string | null) => {
+  const newUrl = url ? url.toLowerCase().trim() : "";
```

Comment:
This function could be less brittle. I'd suggest using the `URL` api to get the parts of the path based on the URL

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#examples